### PR TITLE
Cgmes update simple update chlog index

### PR DESCRIPTION
**Pull request by kaltakovae**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix

**What is the current behavior? (You can also link to an open issue here)**
The Changelog is not ordered: base changes are placed before variant specific changes.

**What is the new behavior (if this is a feature change)?**
This feature implements sequential indexing of network Changelog, according to the order they have been made.

To verify the ordering functionality the unit test does the following:
- make variant specific changes
- make base changes 
- check the order

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/Changelog.java [hunk 1]
```diff
@@ -80,11 +81,13 @@ public void onVariantCreated(String sourceVariantId, String targetVariantId) {
 
     public List<IidmChange> getChangesForVariant(String variantId) {
         if (!changesByVariant.containsKey(variantId)) {
+            // If we only have baseChanges we assume they are already ordered
             return Collections.unmodifiableList(baseChanges);
         } else {
             List<IidmChange> cs = new ArrayList<>(baseChanges.size() + changesByVariant.size());
             cs.addAll(baseChanges);
             cs.addAll(changesByVariant.get(variantId));
+            cs.sort(Comparator.comparing(IidmChange::getIndex));
```
Comment (mathbagu) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/Changelog.java]: If the order is important, maybe we should use a TreeSet, with a comparator on the index?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/IidmChange.java [hunk 2]
```diff
@@ -18,11 +19,18 @@
...
         return identifiable;
     }
 
+    public int getIndex() {
+        return index;
+    }
+
     private final Identifiable identifiable;
+    private final int index;
+    private static AtomicInteger counter = new AtomicInteger();
```
Comment (mathbagu) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/IidmChange.java]: Should be declared as final

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogCreationRemovalTest.java [hunk 3]
```diff
@@ -85,15 +86,15 @@ public void testChangelogRemove() {
...
+        List<IidmChange> list = new ArrayList<>();
         for (IidmChange i : changes) {
             if (!(i instanceof IidmChangeRemoval)) {
                 continue;
             }
-            count++;
+            list.add(i);
         }
-        return count;
+        return list;
```
Comment (mathbagu) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogCreationRemovalTest.java]: Use a stream: it's more modern:
```
return changes.stream().filter(IidmChangeRemoval.class::isInstance).collect(Collectors.toList());
```

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java [hunk 4]
```diff
@@ -104,6 +107,32 @@ public void testCloneVariantVariantSpecificChanges() {
...
     }
 
+    @Test
+    public void testChangesLogRightOrder() {
+        String variant1 = network.getVariantManager().getWorkingVariantId();
+        // Make changes, on Variant and base.
+        makeNetworkChangesVariantSpecific(network);
+        makeRemoveChanges(network);
+        List<IidmChange> changes =  new ArrayList<>(changelog.getChangesForVariant(variant1));
+        assertTrue(changes.size() > 1);
```
Comment (mathbagu) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java]: Do you wan simply to check if the list is empty?

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java [hunk 5]
```diff
@@ -123,6 +152,10 @@ private void makeNetworkChangesVariantSpecific(Network network) {
         load.setP0(newP).setQ0(newQ).getTerminal().setP(newP).setQ(newQ);
     }
 
+    private void makeRemoveChanges(Network network) {
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java]: this method (and other utilitaries test methods) can be static

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/Changelog.java (hunk 1 above)
Comment (kaltakovae) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/Changelog.java]: Done

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogCreationRemovalTest.java (hunk 3 above)
Comment (kaltakovae) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogCreationRemovalTest.java]: Done

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java (hunk 5 above)
Comment (kaltakovae) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java]: Done

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java (hunk 4 above)
Comment (kaltakovae) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/update/ChangelogUpdateTest.java]: To test the right ordering, the list of changes should have multiple items.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/IidmChange.java (hunk 2 above)
Comment (kaltakovae) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/IidmChange.java]: Done

**Comment by MioRtia** [comment]
Could you please correct the Code Smell found by Sonarcloud?

**Pull request merged**